### PR TITLE
[AC][SUPP-279] audit eligibility

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/util/TraceIdExt.scala [hunk 1]
```diff
@@ -34,7 +35,7 @@ object TraceIdExt {
     traceIdStringsFromQueryParameter()
       .flatMap { seq =>
         val set = seq.map(TraceId.apply).toSet
-        if (set.size > 1) Logger(this.getClass).error(s"Multiple traceIds in the URL. [${set.mkString(", ")}]")
+        if (set.size === 1) Logger(this.getClass).error(s"Multiple traceIds in the URL. [${set.mkString(", ")}]")
```
Comment (darrell-plant) [inline app/util/TraceIdExt.scala:38]: this looks wrong

Comment (AndyHWChung) [reply app/util/TraceIdExt.scala:38]: whoops! I changed it cos of a wartremover warning and then didn't need it after all and changed it back to the wrong thing 😬 I'll change it back

File: app/models/audit/eligibility/EligibilityCheck.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.audit.eligibility
+
+import essttp.rootmodel.ttp.ChargeTypeAssessment
+import play.api.libs.json.{Json, OWrites}
+
+final case class EligibilityCheck(
```
Comment (barnesjake) [inline app/models/audit/eligibility/EligibilityCheck.scala:22]: I know it's in the audit directory which I think is good enough, but maybe renaming this to EligibilityCheckAuditDetail or something - just to be crystal clear. I'm not overly fussed though, so feel free to completely ignore 😄

File: app/services/AuditService.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  // TODO: hook in when questions about business rules clarified
+  @silent
+  private def toEligibilityCheck(
+      journey:          ComputedTaxId,
+      enrollmentReason: Either[EnrollmentReasons.NotEnrolled, EnrollmentReasons.InactiveEnrollment]
+  )(implicit r: AuthenticatedJourneyRequest[_]): EligibilityCheck = {
+    EligibilityCheck(
+      EligibilityResult.Ineligible,
```
Comment (darrell-plant) [inline app/services/AuditService.scala:50]: named parameters might be helpful (and below)

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -3,23 +3,23 @@ import play.core.PlayVersion
 
 object AppDependencies {
```
Comment (barnesjake) [inline project/AppDependencies.scala:4]: Nice, thanks for doing these 😄

File: app/models/audit/eligibility/EligibilityCheck.scala (hunk 2 above)
Comment (AndyHWChung) [reply app/models/audit/eligibility/EligibilityCheck.scala]: 👍 clearer is usually better - updated [here](https://github.com/hmrc/essttp-frontend/pull/92/commits/2494869cf8e983d94983c15f96ec8a49d8ff40a2)

File: app/services/AuditService.scala (hunk 3 above)
Comment (AndyHWChung) [reply app/services/AuditService.scala]: 👍 added it [here](https://github.com/hmrc/essttp-frontend/pull/92/commits/2494869cf8e983d94983c15f96ec8a49d8ff40a2) looks clearer now I think

**Review by darrell-plant — APPROVED** [review]

**Review by barnesjake — APPROVED** [review]

**Pull request merged**